### DISCUSSION TITLE
Xiaofang cameras

### DISCUSSION THREAD

**Discussion by clee604** [gtxaspec/wz_mini_hacks#83] [Development]
Hi there,

Many thanks for this, I have it on for rtsp and lan adapter on my wyze v2 and v3 cams and it works like a charm.

Two questions though: 

1. Would this be able to be ported over to the old cheap xiaofang cameras? I believe it uses similar hardware to the wyze v2. There are the fang-hacks here https://github.com/samtap/fang-hacks that still work for me to get rtsp feeds but no ethernet support.
2. Are there any plans to include support beyond FAT32 (which maxes out at 32gb cards) or is this a hard limitation that cannot be worked around? I haven't spent the time to figure out how to record directly to a network share so I still record directly on the SD card for the time being and 32gb isn't a lot.

**Comment by gtxaspec** [comment]
fang hacks for the xiaofang would be equivalent to the v1 camera , which uses a different processor than the v2, won't be supported by wz_mini.  

now, if your xiaofang uses a t20 processor, then it should be supported by the dafang hacks project too.

**Comment by claudobahn** [comment]
@clee604 the 32GB limit for FAT32 seems to be primarily a Windows UI limitation. On Mac/Linux it is possible.

The easiest way around that limitation on Windows, I believe, is to use the command prompt (probably need to run the command prompt / powershell as Administrator) and use something like `format /FS:FAT32 X:`, replace the driver letter `X:` with what's correct for you, obviously.

**Reply by virmaior** [reply]
How do you format FAT32 for > 32 GB cards on OSX? My mac refuses to read the cards that were running fine in the cameras. Does that mean they got reformatted as EXFAT by the cameras at some point?

**Comment by clee604** [comment]
@gtxaspec thank you for the quick answer
@claudobahn nice! I'll try out your command or just use linux to format a larger sd card to test

**Comment by gtxaspec** [comment]
is exfat supported with the v2? I can't recall.

edit: never mind, I forgot uboot doesn't support exfat anyway lol

**Comment by clee604** [comment]
The v2 supports exfat with the newer firmwares I believe, but my main issue was the size limitation which seems to be a windows UI issue anyways
edit: my first attempt failed within windows, may have to use linux or 3rd party software
![image](https://user-images.githubusercontent.com/93049965/173460679-f75bcf91-f900-4496-b6f4-e89b532ba4a4.png)

**Reply by claudobahn** [reply]
@clee604 not sure how high you can go, but I'm currently using 64GB FAT32.

@gtxaspec bummer about uboot, did wyze update their published uboot sources by any chance to include support?

Looking at https://support.wyze.com/hc/en-us/articles/360024852172-Release-Notes-Firmware
* V3 got exFAT w/ 4.36.8.15
* V2 doesn't explicitly spell it out, but maybe 4.9.8.746 which says "Added support for microSD cards larger than 32GB"?

**Reply by gtxaspec** [reply]
@claudobahn, they just updated/added an exfat kernel module so far.  no updates to uboot.

**Reply by clee604** [reply]
I can confirm 128gb works in FAT32 on a V3 with the hacks installed. I had to format the sd card using Linux as windows wouldn't let me without using 3rd party tools.
